File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int result=0;
        for(int e=0,f=height.length-1;e<f;){
            int g=Math.min(height[e],height[f]);
            int h=g*(f-e);
            if(h>result) result=h;
            if(g==height[e]) e++;
            else f--;
        }
        return result;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int result=INT_MIN;
        int e=0;
        int f=height.size()-1;
        while(e<f){
            result=max(result,(f-e)*((height[e]<height[f])?height[e]:height[f]));
            (height[e]<height[f])?e++:f--;
        }
        return result;
    }
};